addons/website_hr_recruitment/models/hr_recruitment_source.py

```diff
@@ -0,0 +1,23 @@
+from urlparse import urljoin
+from werkzeug import url_encode
+
+from openerp import models, fields, api
+from openerp.addons.website.models.website import slug
+
+class HrRecruitmentSource(models.Model):
+    _inherit = 'hr.recruitment.source'
+
+    @api.depends('source_id', 'source_id.name')
+    @api.one
+    def _compute_url(self):
+        base_url = self.env['ir.config_parameter'].get_param('web.base.url')
+        for source in self:
+            source.url = urljoin(base_url, "/jobs/detail/%s?%s" % (slug(source.job_id),
```

Comment:
The hr.job model inherits website.published.mixin (https://github.com/odoo/odoo/blob/master/addons/website_hr_recruitment/models/hr_job.py#L8), which has a website_url field (https://github.com/odoo/odoo/blob/master/addons/website/models/website.py#L809) which calls _website_url (https://github.com/odoo/odoo/blob/master/addons/website/models/website.py#L805) which is overriden by each model inheriting this mixin, including hr.job (https://github.com/odoo/odoo/blob/master/addons/website_hr_recruitment/models/hr_job.py#L13). You should then use this field instead of redefining the url here. Additionally, this method is missing an @api.depends on job_id since chancing the job_id will change the url.